src/Integration.Vsix/CFamily/CLangAnalyzer.cs

```diff
@@ -159,18 +158,17 @@ private void RunAnalysis(IRequest request, IIssueConsumer consumer, IAnalysisSta
         {
             var analysisStartTime = DateTime.Now;
             statusNotifier?.AnalysisStarted(request.Context.File);
-            int issueCount = 0;
 
-            var handleMessage = consumer == null
-                ? (Action<Message>) (message => { })
-                : message => HandleMessage(message, request, consumer, ref issueCount);
+            var messageHandler = consumer == null
+                ? NoOpMessageHandler.Instance
+                : new MessageHandler(request, consumer, issueConverter);
```

Comment:
The only caller that supplies a null consumer is the PCH event listener.
Perhaps I should have changed the PCH event listener to pass a `NoOpIssueConsumer` instead and thrown here if the consumer was null.